### PR TITLE
Delete aws instance

### PR DESCRIPTION
**Pull request by cdiop**
Ensure that the AWS instance is deleted when all the nodes are deleted or the NS is delete

### CODE REVIEW

File: infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java [hunk 1]
```diff
@@ -378,12 +383,46 @@ public void removeNode(Node node) throws RMException {
...
+            nodesPerInstance = (Map<String, Set<String>>) persistedInfraVariables.get(NODES_PER_INSTANCES_KEY);
+            // Make modifications to the nodesPerInstance map
+            if (nodesPerInstance.get(instanceTag) != null) {
+                nodesPerInstance.get(instanceTag).remove(nodeName);
+                logger.info("Removed node: " + nodeName);
+                if (nodesPerInstance.get(instanceTag).isEmpty()) {
+                    logger.info("Instance :" + instanceTag + " is empty ");
+                    if (terminateInstanceIfEmpty) {
+                        logger.info("Call terminate instance for: " + instanceTag);
+                        connectorIaasController.terminateInstance(infrastructureId, instanceTag);
```
Comment (tao-xinxiu) [inline infrastructures/infrastructure-aws-ec2/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AWSEC2Infrastructure.java]: I think you should use `terminateInstanceByTag` as you are using "instanceTag" returned by `getInstanceIdProperty`.

**Comment by cdiop** [comment]
I will do some changes
This call return node.getProperty(INSTANCE_TAG_NODE_PROPERTY) gives the Id of the VM and not the tag for AWS

**Comment by tao-xinxiu** [comment]
> I will do some changes
> This call return node.getProperty(INSTANCE_TAG_NODE_PROPERTY) gives the Id of the VM and not the tag for AWS

That's because you've also passed `INSTANCE_TAG_NODE_PROPERTY` instead of `INSTANCE_ID_NODE_PROPERTY` in `linuxInitScriptGenerator.buildScript`. When RM gets the instance information from a node, it uses what's passed in the node start script.
You can modify them both to be consistent.

**Comment by cdiop** [comment]
> > I will do some changes
> > This call return node.getProperty(INSTANCE_TAG_NODE_PROPERTY) gives the Id of the VM and not the tag for AWS
> 
> That's because you've also passed `INSTANCE_TAG_NODE_PROPERTY` instead of `INSTANCE_ID_NODE_PROPERTY` in `linuxInitScriptGenerator.buildScript`. When RM gets the instance information from a node, it uses what's passed in the node start script.
> You can modify them both to be consistent.

**Pull request merged**